index.js

```diff
@@ -105,23 +113,86 @@ function initApp(callback) {
     /*
      * configure routes
      */
-    app.post('/api/v1/substitution', (req, res) => {
-      if (!req.isAuthenticated() || req.user.level < config.user.level.substitute) {
-        res.status(401).send('{"error":"not authenticated or not allowed"}');
-        return;
-      }
+var Compendium = require('./lib/model/compendium');
+
+    // get all compendia for list in authorView.html
+    app.get('/api/v1/substitution/all', (req, res) => {
+      // if (!req.isAuthenticated() || req.user.level < config.user.level.substitute) {
+      //   res.status(401).send('{"error":"not authenticated or not allowed"}');
+      //   return;
+      // }
+      console.log("GET - all erc");
 
       // TODO implement ...
+      var answer = {};
+      var filter = {};
+      var limit = parseInt(req.query.limit || config.list_limit, 10);
+      var start = parseInt(req.query.start || 1, 10) - 1;
+
+      // add query element to filter (used in database search) and to the query (used for previous/next links)
+      // eslint-disable-next-line no-eq-null, eqeqeq
+      if (req.query.job_id != null) {
+        filter.job_id = req.query.job_id;
+      }
+      // eslint-disable-next-line no-eq-null, eqeqeq
+      if (req.query.user != null) {
+        filter.user = req.query.user;
+      }
+
+      Compendium.find(filter).select('id').skip(start).limit(limit).exec((err, comps) => {
+        if (err) {
+          res.status(500).send(JSON.stringify({ error: 'query failed' }));
+        } else {
+          var count = comps.length;
+          if (count <= 0) {
+            res.status(404).send(JSON.stringify({ error: 'no compendium found' }));
+          } else {
+
+            answer.results = comps.map(comp => {
+              return comp.id;
+            });
+            console.log(answer); // log compendia IDs
+            res.status(200).send(JSON.stringify(answer));
+          }
+        }
+      });
     });
-    
-    app.get('/api/v1/compedium/:id/substition', function (req, res) {
+
+    // get data path of two choosen erc's
+    app.get('/api/v1/substitution/get/:xch1/:xch2', function (req, res) {
       res.setHeader('Content-Type', 'application/json');
-      
-      // query substitions for given id from database
 
-      // create response
-      var response = {};
-      res.send(response);
+      var id_erc1 = req.params.xch1;
+      var id_erc2 = req.params.xch2;
+      var answer = {
+        erc1 : {
+          id : id_erc1,
+          files : []
+        },
+        erc2 : {
+          id : id_erc2,
+          files : []
+        }
+         };
+      // datapath of first erc
+      var folderPath1 = config.fs.compendium + id_erc1 + '/data';
+      try {
+        fs.accessSync(folderPath1); // throws if does not exist
+        answer.erc1.files = dirTree(folderPath1);
+      } catch (e) {
+        res.status(500).send({ error: 'internal error: could not read compendium contents from storage', e });
```

Comment:
don't simply return caught exceptions, because the might contain server paths we do not want the user to see.

instead _log_ the exception, and also add to the log _which of the paths_ gives an error.